testsuite/features/step_definitions/https_connection_steps.rb

```diff
@@ -0,0 +1,84 @@
+# Copyright 2017-2025 SUSE LLC.
+# Licensed under the terms of the MIT license.
+
+### SSL connection checks to server
+
+require 'open-uri'
+require 'uri'
+require 'openssl'
+
+When(/^I connect to the server insecurely$/) do
+  @url = "#{Capybara.app_host}".gsub("https:", "http:")
+  begin
+    @res = URI.open(@url, redirect: false, open_timeout: 5)
+  rescue => @e
+  end
+end
+
+Then(/^the connection should redirect to the secured channel$/) do
+  if not (@e.instance_of?(OpenURI::HTTPRedirect) and @e.message == "302 302") then
+    print("\tReturn code: #{@res.status[0]}\n\tURI: #{@res.base_uri.to_s}\n") if @res
```

Comment:
We don't use `print` in Ruby functions but `puts`.
Still, for Step definitions, as they are processed by Cucumber framework, we recommend to use `log` as that way the line it gonna appear on the Cucumber report. Otherwise, it's only printed on the console output.